scripts/pr_compliance.py

```diff
@@ -0,0 +1,487 @@
+            elif line == "### Change Logs" or line.startswith("**Risk level:") or line == "### Contributor's checklist":
+                print_error(f"Out of order section or missing description\"{line}\"")
+                return False
+            # make sure there is additional text before looking for risk level
+            elif found and line != "_Describe any public API or user-facing feature change or any performance impact._":
+                # this is the default so we want to see different text that is actually explaining things
+                state = "RISKLEVEL"
+                found = False
+
+        elif state == "RISKLEVEL":
+            if line.startswith("**Risk level:"):
+                #if we already saw this then they shouldn't have it again
+                if found:
+                    print_error("duplicate line starting with \"**Risk level:\" ")
+                    return False
+                #if it is this line then they never chose a risk level
+                if line == "**Risk level: none | low | medium | high**":
+                    print_error("risk level not chosen ")
+                    return False
+
+                # an explanation is not required for none or low
+                if "NONE" in line.upper() or "LOW" in line.upper():
+                    state = "CHECKLIST"
+                    found = False
+                elif "MEDIUM" in line.upper() or "HIGH" in line.upper():
+                    # an explanation is required so we don't change state
+                    found = True
+                else:
+                    #they put something weird in for risk level
+                    print_error("invalid choice for risk level")
+                    return False
+            elif line == "### Impact" or line == "### Change Logs" or line == "### Contributor's checklist":
+                print_error(f"Out of order section or missing description \"{line}\"")
+                return False
+
+            elif found and line != "_Choose one. If medium or high, explain what verification was done to mitigate the risks._":
+                #explanation found so we can change states
+                state = "CHECKLIST"
+                found = False
+
+
+        elif state == "CHECKLIST":
+            if line.startswith("**Risk level:"):
+                print_error("duplicate line starting with \"**Risk level:\" ")
+                return False
+            if line == "### Impact" or line == "### Change Logs" or line.startswith("**Risk level:"):
+                print_error(f"Out of order section \"{line}\"")
+                return False
+            if line == "### Contributor's checklist":
+                #this has passed the check
+                return True
+    #we didn't exit(0) so something is wrong with the template
+    print_error("template is not filled out properly")
+    return False
+
+def test_body():
+    debug_messages = True
+    just_the_template = [
```

Comment:
you could load this from https://github.com/apache/hudi/blob/master/.github/PULL_REQUEST_TEMPLATE.md ? so that you don't need to duplicate it